crates/kitsune-db/migrations/2023-12-16-192941_initial/up.sql

```diff
@@ -0,0 +1,771 @@
+        UNIQUE (url);
+
+ALTER TABLE accounts_follows
+    ADD CONSTRAINT "uk-accounts_follows-account_id-follower_id"
+        UNIQUE (account_id, follower_id);
+
+-- Foreign key constraints
+ALTER TABLE accounts_follows
+    ADD CONSTRAINT "const-foreign-accounts_follows-account_id"
+        FOREIGN KEY (account_id) REFERENCES accounts (id) ON DELETE CASCADE ON UPDATE CASCADE;
+
+ALTER TABLE accounts_follows
+    ADD CONSTRAINT "fk-accounts_follows-follower_id"
+        FOREIGN KEY (follower_id) REFERENCES accounts (id) ON DELETE CASCADE ON UPDATE CASCADE;
+
+-- Create indexes
+
+CREATE INDEX "idx-accounts_follows-account_id" ON accounts_follows (account_id);
+CREATE INDEX "idx-accounts_follows-follower_id" ON accounts_follows (follower_id);
+
+-- Register triggers
+
+SELECT diesel_manage_updated_at('accounts_follows');
+
+-- END "accounts_follows" TABLE
+
+-- BEGIN "accounts_preferences" TABLE
+
+CREATE TABLE accounts_preferences
+(
+    account_id               UUID PRIMARY KEY,
+    notify_on_follow         BOOLEAN NOT NULL,
+    notify_on_follow_request BOOLEAN NOT NULL,
+    notify_on_repost         BOOLEAN NOT NULL,
+    notify_on_post_update    BOOLEAN NOT NULL,
+    notify_on_favourite      BOOLEAN NOT NULL,
+    notify_on_mention        BOOLEAN NOT NULL
+);
+
+-- Foreign key constraints
+ALTER TABLE accounts_preferences
+    ADD CONSTRAINT "fk-accounts_preferences-account_id"
+        FOREIGN KEY (account_id)
+            REFERENCES accounts (id)
+            ON DELETE CASCADE
+            ON UPDATE CASCADE;
+
+-- END "accounts_preferences" TABLE
+
+-- BEGIN "users_accounts" TABLE
+
+CREATE TABLE users_accounts
+(
+    user_id    UUID,
+    account_id UUID,
+    PRIMARY KEY (user_id, account_id)
+);
+
+-- Foreign key constraints
+ALTER TABLE users_accounts
+    ADD CONSTRAINT "fk-users_accounts-user_id"
+        FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE CASCADE ON UPDATE CASCADE;
+
+ALTER TABLE users_accounts
+    ADD CONSTRAINT "fk-users_accounts-account_id"
+        FOREIGN KEY (account_id) REFERENCES accounts (id) ON DELETE CASCADE ON UPDATE CASCADE;
+
+-- END "users_accounts" TABLE
+
+-- BEGIN "link_previews" TABLE
+
+CREATE TABLE link_previews
+(
+    url        TEXT PRIMARY KEY,
+    embed_data JSONB       NOT NULL,
+    expires_at TIMESTAMPTZ NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+-- Register triggers
+
+SELECT diesel_manage_updated_at('link_previews');
+
+-- END "link_previews" TABLE
+
+-- BEGIN "posts" TABLE
+
+CREATE TABLE posts
+(
+    id               UUID PRIMARY KEY,
+    account_id       UUID                                                                                                   NOT NULL,
+
+    in_reply_to_id   UUID,
+    reposted_post_id UUID,
+
+    is_sensitive     BOOLEAN                                                                                                NOT NULL,
```

Comment:
Misskey sets the `sensitive` flags on a per-medium basis, i.e. authors can attach non-sensitive media and sensitive media to a single post at the same time. Is this a use case Kitsune would like to support?

Reply:
Good idea actually.. and for Mastodon, where we have to have a single sensitive flag, we simply do an OR chain for the `is_sensitive` (or an `.any` on an iterator, in Rust terms :P)